provider/go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/pulumi/pulumi-aws-native/provider
 
-go 1.21
+go 1.21.0
```

Comment:
curious, why pin to 1.21.0?

Reply:
I didn't make this change - I think this was done by `go mod tidy` 🤷